extensions/arc/src/ui/tree/controllerTreeNode.ts

```diff
@@ -38,10 +39,19 @@ export class ControllerTreeNode extends TreeNode {
 			await this.model.refresh(false);
 			await this._childrenRefreshPromise.promise;
 		} catch (err) {
-			// Couldn't get the children and TreeView doesn't have a way to collapse a node
-			// in a way that will refetch its children when expanded again so instead we
-			// display a tempory node that will prompt the user to re-enter credentials
-			return [new RefreshTreeNode(this)];
+			vscode.window.showErrorMessage(loc.errorConnectingToController(err));
+			try {
+				await this.model.refresh(false, true);
+				await this._childrenRefreshPromise.promise;
```

Comment:
Is the idea here to give the user a 2nd chance to enter correct credentials?

Reply:
Sorta - this is mostly for the case where you had a node already registered but then something happened and you can't connect (can't reach server, username/password changed, etc). So in that case when they try to expand the node normally it would be able to get the children just fine but in this case we can't reach it so we re-prompt them for credentials in case it's something they can solve on their own.